lib/jira/commands/transition.rb

```diff
@@ -58,14 +63,42 @@ def transitions
         @transitions ||= (
           transitions = {}
           metadata['transitions'].each do |transition|
-            transitions[transition['to']['name']] = transition['id']
+            transitions[transition['to']['name']] = {
+              id: transition['id'],
+              "fields"=> { "resolution"=> { "required"=> resolution_required?(transition) } }
+            }
           end
           transitions
         )
       end
 
       def metadata
-        @metadata ||= api.get("issue/#{ticket}/transitions")
+        @metadata ||= api.get("issue/#{ticket}/transitions?expand=transitions.fields")
+      end
+
+      def resolution
+        @resolution ||= resolutions[
+          options['resolution'] || io.select("Resolve #{ticket} as:", resolutions.keys)
+        ]
+      end
+
+      def resolutions
+        @resolutions ||= (
+          resolutions = {}
+          metadata_resolutions.each do |resolution|
+            resolutions[resolution['name']] = resolution['name']
+          end
+          resolutions
+        )
+      end
+
+      def resolution_required?(transition)
+        return false if transition['fields']['resolution'].nil?
+        transition['fields']['resolution']['required'] || false
+      end
+
+      def metadata_resolutions
```

Comment:
since `metadata_resolutions` is not being used again, it probably doesnt need to be memoized or a separate method
can probably just do `api.get("resolution").each` in the resolutions method